samples/web/content/apprtc/room.py

```diff
@@ -0,0 +1,422 @@
+    mandatory = (tokens[0] == 'mandatory')
+  else:
+    # Otherwise, default to mandatory, except for goog constraints, which
+    # won't work in other browsers.
+    mandatory = not tokens[0].startswith('goog')
+
+  tokens = tokens[-1].split('=')
+  if len(tokens) == 2:
+    if mandatory:
+      track_constraints['mandatory'][tokens[0]] = tokens[1]
+    else:
+      track_constraints['optional'].append({tokens[0]: tokens[1]})
+  else:
+    logging.error('Ignoring malformed constraint: ' + constraint_string)
+
+def make_media_track_constraints(constraints_string):
+  if not constraints_string or constraints_string.lower() == 'true':
+    track_constraints = True
+  elif constraints_string.lower() == 'false':
+    track_constraints = False
+  else:
+    track_constraints = {'mandatory': {}, 'optional': []}
+    for constraint_string in constraints_string.split(','):
+      add_media_track_constraint(track_constraints, constraint_string)
+
+  return track_constraints
+
+def make_media_stream_constraints(audio, video, firefox_fake_device):
+  stream_constraints = (
+      {'audio': make_media_track_constraints(audio),
+       'video': make_media_track_constraints(video)})
+  if firefox_fake_device:
+    stream_constraints['fake'] = True
+  logging.info('Applying media constraints: ' + str(stream_constraints))
+  return stream_constraints
+
+def maybe_add_constraint(constraints, param, constraint):
+  if (param.lower() == 'true'):
+    constraints['optional'].append({constraint: True})
+  elif (param.lower() == 'false'):
+    constraints['optional'].append({constraint: False})
+
+  return constraints
+
+def make_pc_constraints(dtls, dscp, ipv6):
+  constraints = { 'optional': [] }
+  maybe_add_constraint(constraints, dtls, 'DtlsSrtpKeyAgreement')
+  maybe_add_constraint(constraints, dscp, 'googDscp')
+  maybe_add_constraint(constraints, ipv6, 'googIPv6')
+
+  return constraints
+
+def append_url_arguments(request, link):
+  arguments = request.arguments()
+  if len(arguments) == 0:
+    return link
+  link += ('?' + cgi.escape(arguments[0], True) + '=' +
+           cgi.escape(request.get(arguments[0]), True))
+  for argument in arguments[1:]:
+    link += ('&' + cgi.escape(argument, True) + '=' +
+             cgi.escape(request.get(argument), True))
+  return link
+
+def get_wss_parameters(request):
+  ws_host_port_pair = request.get('wshpp')
+  ws_tls = request.get('wstls')
+
+  if not ws_host_port_pair:
+    ws_host_port_pair = constants.WSS_HOST_PORT_PAIR
+
+  if ws_tls and ws_tls == 'false':
+    wss_url = 'ws://' + ws_host_port_pair + '/ws'
+    wss_post_url = 'http://' + ws_host_port_pair
+  else:
+    wss_url = 'wss://' + ws_host_port_pair + '/ws'
+    wss_post_url = 'https://' + ws_host_port_pair
+  return (wss_url, wss_post_url)
+
+# Returns appropriate room parameters based on query parameters in the request.
+# TODO(tkchin): move query parameter parsing to JS code.
+def get_room_parameters(request, room_id, client_id, is_initiator):
```

Comment:
I'd keep this one and the other parameter related methods in apprtc.py, since they really have nothing to do with Room. Or put them in a separate file.